advocacy_docs/security/securing-postgresql/301/index.mdx

```diff
@@ -0,0 +1,168 @@
+
+### Health Insurance Portability and Accountability Act (HIPAA)
+
+HIPAA governs the protection of healthcare data in the United States. PostgreSQL installations dealing with protected health information (PHI) must meet stringent confidentiality, integrity, and availability requirements.
+
+- **Encrypt PHI.** All PHI must be encrypted at rest and in transit. Use pgcrypto or external encryption tools to secure PHI in PostgreSQL.
+
+- **Access control.** Implement strong authentication and authorization. Ensure that users and roles are defined clearly, and use multi-factor authentication (MFA) for administrative access.
+
+- **Audit trails.** HIPAA requires tracking and logging any access to PHI. You can configure PostgreSQL’s logging system and pgaudit to log these actions. For example:
+
+```bash
+log_statement = 'all'
+log_min_duration_statement = 1000
+```
+
+## PostgreSQL in FISMA-compliant environments
+
+The Federal Information Security Management Act (FISMA) establishes security requirements for federal IT systems. To be used in FISMA environments, PostgreSQL must comply with the NIST SP 800-53 framework.
+
+- **FIPS-140-2 encryption.** Ensure PostgreSQL uses FIPS-compliant encryption algorithms for secure communication. PostgreSQL can be integrated with OpenSSL configured for FIPS mode, or you can use external encryption tools.
+
+- **Multi-factor authentication (MFA).** Use MFA for administrative access to the database. Integration with external identity providers (for example, Okta, AWS IAM) can help enforce MFA policies for critical roles.
+
+- **Incident response.** Configure PostgreSQL to detect and respond to security incidents. All security-related events must be logged, and systems must have defined incident response plans.
+
+## Advanced encryption practices
+
+Encryption is a cornerstone of advanced database security. Beyond basic encryption of data in transit and at rest, advanced encryption practices include key management and more sophisticated data protection strategies.
+
+### Key management
+
+- **External key management.** Rather than storing encryption keys within the database or filesystem, use external systems like AWS KMS, HashiCorp Vault, or Azure Key Vault to manage encryption keys. For example:
+
+```bash
+aws kms encrypt --key-id alias/myKey --plaintext fileb://myfile
+```
+
+- **Key rotation.** To limit the potential damage from key compromise, regularly rotate encryption keys. Ensure PostgreSQL encryption supports key rotation without downtime.
+  
+### Transparent data encryption (TDE)
+
+TDE encrypts the entire database at the file level. While not natively supported in PostgreSQL, tools like pgcrypto and external software can implement TDE.
+
+- **Use pgTDE.** You can use the pgTDE extension to encrypt entire databases or specific tablespaces. Data is encrypted transparently as it's written to disk.
```

Comment:
What's pgTDE? It sounds like a predecessor of the TDE extensions in PGE and EPAS? pg_tde is an experimental percona extension while pgtde seems to be a VMware propietary extension. pgcrypto is not TDE though, it's column level encryption.

TDE transparently encrypts the entire database at the file level. While not natively supported in PostgreSQL, EDB Postgres Extended Server and EDB Postgres Advanced Server both support TDE. For PostgreSQL itself, extensions are available which can implement TDE.